mirascope/beta/openai/realtime/realtime.py

```diff
@@ -246,33 +359,61 @@ async def _process_receiver(self, conn: ClientConnection) -> None:
+                        )
                         self._text_message_received = True
                     case "response.text.delta":
-                        text_chunk += message["delta"]
+                        text_chunks[message["response_id"]] += message["delta"]
                         await self._process_specific_receiver(message, "text_chunk")
                     case "response.audio_transcript.done":
                         await self._process_specific_receiver(
-                            audio_transcript_chunk, "audio_transcript"
+                            audio_transcript_chunks[message["response_id"]],
+                            "audio_transcript",
                         )
-                        audio_transcript_chunk = ""
                         self._audio_transcript_received = True
                     case "response.audio_transcript.delta":
-                        audio_transcript_chunk += message["delta"]
+                        audio_transcript_chunks[message["response_id"]] += message[
+                            "delta"
+                        ]
                         await self._process_specific_receiver(
-                            audio_transcript_chunk, "audio_transcript_chunk"
+                            message, "audio_transcript_chunk"
                         )
-
-                    # TODO: Add more cases like "response.function_call_arguments.done"
-
+                    case "response.function_call_arguments.delta":
+                        function_call_arguments_chunks[message["response_id"]][
+                            message["call_id"]
+                        ] += message["delta"]
+                        await self._process_specific_receiver(message, "tool_chunk")
+                    case "response.function_call_arguments.done":
+                        response_id = message["response_id"]
+                        tool_schemas = self._temporary_tool_schemas.pop(response_id, [])
+                        if tool_schemas:
+                            tool_schema = tool_schemas.pop(0)
+                        else:
+                            tool_schema = self._tool_schemas[message["call_id"]]
+                        function_call_arguments = function_call_arguments_chunks[
+                            response_id
+                        ].pop(message["call_id"])
+                        if not function_call_arguments_chunks[response_id]:
+                            del function_call_arguments_chunks[response_id]
+                        tool = tool_schema.from_tool_call(
+                            FunctionCallArguments(
+                                call_id=message["call_id"],
+                                arguments=function_call_arguments,
+                            )
+                        )
+                        await self._process_specific_receiver(tool, "tool")
+                        # TODO: type: "function_call_output".
```

Comment:
as mentioned in my previous comment, the user is required to respond with the `function_call_output` event or it won't work, so we have to support this for tools to work at all

see my other comment on an idea for how to handle this in the interface